File: core/ValidParentheses/ValidParentheses.py
def isValid(s: str) -> bool:
    stack = []
    for ch in s:
        if stack and ((ch == ")" and stack[-1] == "(") or (ch == "}" and stack[-1] == "{") or (ch == "]" and stack[-1] == "[")):
            stack.pop()
        else:
            stack.append(ch)

    return len(stack) == 0


if __name__ == '__main__':
    print(isValid("]"))

File: core/ValidParentheses/ValidParentheses.java
package com.dbc;

import java.util.Stack;

public class ValidParentheses {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            if (!stack.isEmpty() && ((stack.peek() == '(' && s.charAt(i) == ')') || (stack.peek() == '{' && s.charAt(i) == '}') || (stack.peek() == '[' && s.charAt(i) == ']')))
                stack.pop();
            else stack.push(s.charAt(i));
        }
        return stack.isEmpty();
    }
}
